framework/PostProcessors/Metric.py

```diff
@@ -152,24 +169,34 @@ def inputToInternal(self, currentInputs):
       elif inputType == 'HDF5':
         self.raiseAnError(IOError, "Input type '", inputType, "' can not be accepted")
       elif inputType == 'PointSet':
-        pass #Allowed type
+        hasPointSet = True
       elif inputType == 'HistorySet':
-        self.dynamic = True
-        self.raiseAnError(IOError, "Metric can not process HistorySet, because this capability is not implemented yet")
+        hasHistorySet = True
+        if self.multiOutput == 'raw_values':
+          self.dynamic = True
+          if self.pivotParameter not in currentInput.getVars('indexes'):
+            self.raiseAnError(IOError, self, 'Pivot parameter', self.pivotParameter,'has not been found in DataObject', currentInput.name)
+          if not currentInput.checkIndexAlignment(indexesToCheck=self.pivotParameter):
+            self.raiseAnError(IOError, "HistorySet", currentInput.name," is not syncronized, please use Interfaced PostProcessor HistorySetSync to pre-process it")
+          pivotValues = currentInput.asDataset()[self.pivotParameter].values
+          if len(self.pivotValues) == 0:
+            self.pivotValues = pivotValues
+          elif set(self.pivotValues) != set(pivotValues):
+            self.raiseAnError(IOError, "Pivot values for pivot parameter",self.pivotParameter, "in provided HistorySets are not the same")
       else:
         self.raiseAnError(IOError, "Metric cannot process "+inputType+ " of type "+str(type(currentInput)))
+    if self.multiOutput == 'raw_values' and hasPointSet and hasHistorySet:
```

Comment:
```if self.multiOutput == 'raw_values' and hasDataObject:```

Reply:
This will be checked if both PointSet and HistorySet are provided.